src/GovUk.Education.ExploreEducationStatistics.Publisher/Startup.cs

```diff
@@ -64,21 +64,34 @@ public override void Configure(IFunctionsHostBuilder builder)
                 .AddScoped<IContentReleaseService, ContentReleaseService>()
                 .AddScoped<IReleaseFileRepository, ReleaseFileRepository>()
                 .AddScoped<IReleaseCacheService, ReleaseCacheService>()
-
-                .AddScoped<IPublishingService, PublishingService>(provider =>
-                    new PublishingService(
-                        publicStorageConnectionString: GetConfigurationValue(provider, "PublicStorage"),
-                        privateBlobStorageService: GetBlobStorageService(provider, "CoreStorage"),
-                        publicBlobStorageService: GetBlobStorageService(provider, "PublicStorage"),
-                        methodologyService: provider.GetRequiredService<IMethodologyService>(),
-                        publicationRepository: provider.GetRequiredService<IPublicationRepository>(),
-                        releaseService: provider.GetRequiredService<IReleaseService>(),
-                        logger: provider.GetRequiredService<ILogger<PublishingService>>()))
+                .AddScoped<IPublishingService, PublishingService>()
+                .AddScoped<IPublicBlobStorageService, BlobStorageService>(provider =>
+                {
+                    var publicConnectionString = GetConfigurationValue(provider, "PublicStorage");
+                    return new BlobStorageService(
+                        connectionString: publicConnectionString,
+                        client: new BlobServiceClient(publicConnectionString),
+                        logger: provider.GetRequiredService<ILogger<BlobStorageService>>(),
+                        storageInstanceCreationUtil: new StorageInstanceCreationUtil());
+                })
+                .AddScoped<IPrivateBlobStorageService, BlobStorageService>(provider =>
+                {
+                    var privateConnectionString = GetConfigurationValue(provider, "CoreStorage");
+                    return new BlobStorageService(
```

Comment:
Again, with a Public and Private concrete class that each extracted their own connection string and created their own BlobServiceClient in their constructor, you wouldn't have to do all of this in-depth wiring here.